src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -37,13 +37,33 @@ public class WalletRPCController : FeatureController
         /// <summary>Wallet transaction handler.</summary>
         private readonly IWalletTransactionHandler walletTransactionHandler;
 
-        public WalletRPCController(IWalletManager walletManager, IWalletTransactionHandler walletTransactionHandler, IFullNode fullNode, IBroadcasterManager broadcasterManager, ILoggerFactory loggerFactory) : base(fullNode: fullNode)
+        private readonly WalletSettings walletSettings;
+
+        public WalletRPCController(IWalletManager walletManager, 
+            IWalletTransactionHandler walletTransactionHandler, 
+            IFullNode fullNode, 
+            IBroadcasterManager broadcasterManager, 
+            ILoggerFactory loggerFactory,
+            WalletSettings walletSettings) : base(fullNode: fullNode)
         {
             this.walletManager = walletManager;
             this.walletTransactionHandler = walletTransactionHandler;
             this.fullNode = fullNode;
             this.broadcasterManager = broadcasterManager;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.walletSettings = walletSettings;
+
+            if (walletSettings.DefaultWallet)
+            {
+                try
+                {
+                    this.UnlockWallet(walletSettings.DefaultWalletPassword, maxDurationInSeconds);
```

Comment:
What do you think about putting timeout in config instead of hardcoded? Maybe if it's not specified not unlock wallet at all? I'm just thinking there may be value for something in having a default wallet created that is not unlocked for some usecase.

Reply:
That does not sound like a useful feature ;-) If there comes a scenario where default wallet that is locked is useful, I think it should be considered to be implemented then and not now. This covers a use case that is important, while a default wallet that is locked is not something I've heard about or encountered before. If someone has such as scenario, it is likely they can call the RPC methods to get this behavior, or the REST API, don't you think?

Reply:
I think the wallet shouldn't be unlocked by default.
Having a default wallet doesn't mean it should be ready to stake or whatever straight away. 
Even in StratisX, the wallet isn't unlocked for staking straight away and we have to call `walletpassphrase`.
Also, I think my first point is valid: if the RPC stuff is loaded before the wallet, and a user doesn't have a default wallet and starts the node with -defaultwallet, it will fail in `WalletRPCController`'s ctor because there is no wallet to unlock yet. As it happens, the wallet stuff is loaded first, so it works, but it's really coincidental.

Reply:
"I think the wallet shouldn't be unlocked by default."

Could have yet another parameter that auto-unlocks the wallet, but I don't see the benefit and value of doing this, as there is already RPC method to call and unlock wallet after startup.

There are a few arguments to why not having manual unlock:

- Less secure, because the consumer of RPC needs to have the wallet password stored and sent over an completely insecure network connection. With the default password stored on the machine, it i less accessible from the outside.

- If the user can already do calls to unlock the wallet, the point of a default wallet disappears and they can do multiple calls to first create a wallet (it not exists), and then unlock it in another call.

With a default wallet, it becomes possible to perform payments on a docker instance running the full node, without the need for additional scripts / boot procedures.

Bitcoin Core works with an unencrypted default wallet that allows sending of BTC without any passwords. Since Stratis only does encrypted wallets, this default wallet implementation will "simulate" an unecrypted wallet with a default password.

Regarding the startup logic, could that be improved by having feature dependencies set to ensure that the wallet is loaded before RPC is being made available for calls and unlocking?

Reply:
I think the default wallet is something that we need for integration with NBitcoin who requires a default wallet. Otherwise I would have advised against, I don't particularly like the idea.

Do you know if it needs the wallet to be unlocked?
If we can unlock it with another call, that's my preferred option because it stays aligned with stratisX.
@sondreb @mikedennis

Reply:
Yes for the nbitcoin tests I need to be able to unlock it through config. I can't change the tests to make an additional rpc call

Reply:
Ok cool. In this case, let's add an `unlockdefaultwallet` argument that only works with defaultwallet and defaultpassword.
@sondreb @mikedennis Can one of you do it soon? otherwise I can do it today.

Reply:
I have implemented the `unlockdefaultwallet`, renamed the `defaultwalletpassword` to `defaultpassword`. Changed the `defaultwallet` to be name, instead of a boolean flag.

Final issue which I'm unsure how to achieve: Making sure that the WalletRPCController constructor is always called after WalletFeature.InitializeAsync. At least during development and debugging, it was never an issue, but that doesn't mean it might be a problem if someone uses or initializes the controller differently in the future.

Reply:
ok that's great, I'll have a more thorough look tomorrow but it looks good at first glance 💪